### PR TITLE
Enable component custom sources in Provisioner schema

### PR DESCRIPTION
**Pull request by franpog859**
**Description**

Changes proposed in this pull request:

- Enable component custom sources feature in Provisioner schema

GraphQL mutation change example:
```gql
mutation {
   provisionRuntime(
      config: {
         kymaConfig: {
            components: [
               {
                  component: "cluster-essentials"
                  namespace: "kyma-system"
                  sourceURL: "github.com/kyma-project/kyma.git//resources/cluster-essentials" 
               }
...
```

**Related issue(s)**

See the issue #942

### CODE REVIEW

File: components/provisioner/internal/provisioning/input_converter_test.go [hunk 1]
```diff
@@ -498,6 +501,11 @@ func fixKymaGraphQLConfigInput() *gqlschema.KymaConfigInput {
 					fixGQLConfigEntryInput("test.config.key2", "value2", util.BoolPtr(false)),
 				},
 			},
+			{
+				Component: rafterComponent,
+				Namespace: kymaSystemNamespace,
+				SourceURL: util.StringPtr(rafterSourceURL),
+			},
```
Comment (Szymongib) [inline components/provisioner/internal/provisioning/input_converter_test.go]: Can we also add a component with the source URL to resolver integration tests? That way we will verify it works fine with DB,

Comment (franpog859) [reply components/provisioner/internal/provisioning/input_converter_test.go]: All right!

**Comment by franpog859** [comment]
@Szymongib sure! I talked with the team responsible for the nested field in the Installer CR. It was done "just in case" so I think the simple version could be better here, in the Provisioner API

**Pull request merged**